Userland/Libraries/LibCore/SOCKSProxyClient.cpp

```diff
@@ -102,12 +102,14 @@ ErrorOr<void> send_version_identifier_and_method_selection_message(Core::Socket&
         .method_count = 1,
         .methods = { to_underlying(method) },
     };
-    auto size = TRY(socket.write({ &message, sizeof(message) }));
+    // FIXME: This should write the entire span.
+    auto size = TRY(socket.write_some({ &message, sizeof(message) }));
     if (size != sizeof(message))
         return Error::from_string_literal("SOCKS negotiation failed: Failed to send version identifier and method selection message");
```

Comment:
I don't get this FIXME (just a sample).

Reply:
`write_some` is (technically) allowed to stop writing whenever it wants, so just calling that function once and expecting the length of the returned span to equal the length of the passed span to equal will yield false errors. Therefore, we should actually attempt to write the entire span here, not just "try once and die if it doesn't work".

Note that all of the "This should read/write the entire span" FIXMEs will get taken care of in the later commits of this PR.